File: clients/php/Model/Order.php
<?php

namespace Model {
    require_once 'Model/VehicleOrder.php';
    require_once 'Stream.php';

    /**
     * Player's orders
     */
    class Order
    {
        /**
         * TODO - Document
         */
        public array $vehicles;
    
        function __construct(array $vehicles)
        {
            $this->vehicles = $vehicles;
        }
    
        /**
         * Read Order from input stream
         */
        public static function readFrom(\InputStream $stream): Order
        {
            $vehicles = [];
            $vehiclesSize = $stream->readInt32();
            for ($vehiclesIndex = 0; $vehiclesIndex < $vehiclesSize; $vehiclesIndex++) {
                $vehiclesElement = \Model\VehicleOrder::readFrom($stream);
                $vehicles[] = $vehiclesElement;
            }
            return new Order($vehicles);
        }
        
        /**
         * Write Order to output stream
         */
        public function writeTo(\OutputStream $stream): void
        {
            $stream->writeInt32(count($this->vehicles));
            foreach ($this->vehicles as $element) {
                $element->writeTo($stream);
            }
        }
    }
}

File: clients/cpp/model/Order.cpp
#include "Order.hpp"

namespace model {

Order::Order(std::vector<model::VehicleOrder> vehicles) : vehicles(vehicles) { }

// Read Order from input stream
Order Order::readFrom(InputStream& stream) {
    std::vector<model::VehicleOrder> vehicles = std::vector<model::VehicleOrder>();
    size_t vehiclesSize = stream.readInt();
    vehicles.reserve(vehiclesSize);
    for (size_t vehiclesIndex = 0; vehiclesIndex < vehiclesSize; vehiclesIndex++) {
        model::VehicleOrder vehiclesElement = model::VehicleOrder::readFrom(stream);
        vehicles.emplace_back(vehiclesElement);
    }
    return Order(vehicles);
}

// Write Order to output stream
void Order::writeTo(OutputStream& stream) const {
    stream.write((int)(vehicles.size()));
    for (const model::VehicleOrder& vehiclesElement : vehicles) {
        vehiclesElement.writeTo(stream);
    }
}

// Get string representation of Order
std::string Order::toString() const {
    std::stringstream ss;
    ss << "Order { ";
    ss << "vehicles: ";
    ss << "[ ";
    for (size_t vehiclesIndex = 0; vehiclesIndex < vehicles.size(); vehiclesIndex++) {
        const model::VehicleOrder& vehiclesElement = vehicles[vehiclesIndex];
        if (vehiclesIndex != 0) {
            ss << ", ";
        }
        ss << vehiclesElement.toString();
    }
    ss << " ]";
    ss << " }";
    return ss.str();
}

}